bokeh/util/compiler.py

```diff
@@ -119,13 +121,23 @@ def _detect_nodejs():
     for nodejs_path in nodejs_paths:
         try:
             proc = Popen([nodejs_path, "--version"], stdout=PIPE, stderr=PIPE)
-            proc.wait()
+            (stdout, _) = proc.communicate()
         except OSError:
-            pass
-        else:
-            return nodejs_path
+            continue
+
+        if proc.returncode != 0:
+            continue
+
+        match = re.match(r"^v(\d+)\.(\d+)\.(\d+).*$", stdout.decode("utf-8"))
+
+        if match is not None:
+            version = tuple(int(v) for v in match.groups())
+
+            if version >= nodejs_min_version:
+                return nodejs_path
     else:
-        raise RuntimeError('node.js is needed to allow compilation of custom models ' +
+        version = ".".join(map(str, nodejs_min_version))
+        raise RuntimeError('node.js v%s or higher is needed to allow compilation of custom models ' % version +
                            '("conda install -c bokeh nodejs" or follow https://nodejs.org/en/download/)')
```

Comment:
I don't  an obscure `for/else` is needed here. If a match with a sufficient version is ever found, a *return* happens immediately. Otherwise if the loop exits at all, one was not found so the exception can be raised unconditionally. `for/else` would be only be needed it the loop was `break`ing instead of `return`int

Reply:
@mattpap do you object if I make this change?

Reply:
@bryevdv, go ahead. This for-else may be a leftover from an earlier implementation.